TOOLS/dylib_unhell.py

```diff
@@ -41,83 +41,106 @@ def otool(objfile, rapths):
+        for line in result.splitlines()
+    ]
 
 def resolve_lib_path(objfile, lib, rapths):
     if os.path.exists(lib):
         return lib
 
-    if lib.startswith('@rpath/'):
-        lib = lib[len('@rpath/'):]
+    if lib.startswith("@rpath/"):
+        lib = lib[len("@rpath/"):]
         for rpath in rapths:
             lib_path = os.path.join(rpath, lib)
             if os.path.exists(lib_path):
                 return lib_path
-    elif lib.startswith('@loader_path/'):
-        lib = lib[len('@loader_path/'):]
+    elif lib.startswith("@loader_path/"):
+        lib = lib[len("@loader_path/"):]
         lib_path = os.path.normpath(os.path.join(objfile, lib))
         if os.path.exists(lib_path):
             return lib_path
 
-    raise Exception('Could not resolve library: ' + lib)
+    raise Exception("Could not resolve library: " + lib)
 
 def check_vulkan_max_version(version):
     try:
-        result = subprocess.check_output("pkg-config vulkan --max-version=" + version, shell = True)
+        cmd = f"pkg-config vulkan --max-version={version}"
+        subprocess.check_output(cmd, shell=True)
         return True
-    except:
+    except Exception:
         return False
 
 def get_homebrew_prefix():
-    # set default to standard ARM path, intel path is already in the vulkan loader search array
+    # set default to standard ARM path, intel path is already in the vulkan
+    # loader search array
     result = "/opt/homebrew"
     try:
-        result = subprocess.check_output("brew --prefix", universal_newlines=True, shell=True, stderr=subprocess.DEVNULL).strip()
-    except:
+        result = subprocess.check_output(
+            ["brew", "--prefix"],
+            universal_newlines=True,
+            stderr=subprocess.DEVNULL,
+        ).strip()
+    except Exception:
         pass
 
     return result
 
 def install_name_tool_change(old, new, objfile):
-    subprocess.call(["install_name_tool", "-change", old, new, objfile], stderr=subprocess.DEVNULL)
+    subprocess.call(
+        ["install_name_tool", "-change", old, new, objfile],
+        stderr=subprocess.DEVNULL,
+    )
 
 def install_name_tool_id(name, objfile):
-    subprocess.call(["install_name_tool", "-id", name, objfile], stderr=subprocess.DEVNULL)
+    subprocess.call(
+        ["install_name_tool", "-id", name, objfile],
+        stderr=subprocess.DEVNULL,
+    )
 
 def install_name_tool_add_rpath(rpath, binary):
     subprocess.call(["install_name_tool", "-add_rpath", rpath, binary])
 
 def install_name_tool_delete_rpath(rpath, binary):
     subprocess.call(["install_name_tool", "-delete_rpath", rpath, binary])
 
-def libraries(objfile, result = dict(), result_relative = set(), rapths = []):
+def libraries(objfile, result=None, result_relative=None, rapths=None):
+    if result is None:
+        result = {}
+
+    if result_relative is None:
+        result_relative = set()
+
+    if rapths is None:
+        rapths = []
```

Comment:
changing from default values in the function arguments to explicit checks in the function body was done in `ci/lint: replace format strings`. is this some personal preference since it doesn't really correspond to the commit message.

Reply:
I've updated the error message to make it more relevant. In this case the previous code seems like a bug since default values are shared global values. So multiple calls will reuse the same dict/set/list.
```
>>> def add_to_list(item, items=[]):
...     items.append(item)
...     return items
...
>>> add_to_list(1)
[1]
>>> add_to_list(1)
[1, 1]
```

By using a `None` default, we can force a new dict/set/list for each call to this function.